darts/models/forecasting/forecasting_model.py

```diff
@@ -885,6 +960,14 @@ def historical_forecasts(
                     # use retrained model if `retrain` is not training every step
                     model = model if model is not None else self
 
+                    # slice the series for prediction without retraining
+                    if min_timestamp_predict > series_.time_index[0]:
+                        train_series = series_.drop_before(
+                            min_timestamp_predict - 1 * series_.freq
+                        ).drop_after(pred_time)
+                    else:
+                        train_series = series_.drop_after(pred_time)
```

Comment:
no need to drop before, right?
```suggestion
                    train_series = series_.drop_after(pred_time)
```

Reply:
In that case could we try to do this only once per time series, and not in every fit/predict iteration?

Reply:
If we alternate between "training" and "predict" mode, the `drop _before` needs to be called at each iteration...

Reply:
You're correct, since we overwrite `train_series` at each iteration, it should not interfere with the training mode and the predict mode is indeed capable of extracting the required inputs.